java.hints/src/org/netbeans/modules/java/hints/suggestions/ConvertToVarHint.java

```diff
@@ -0,0 +1,449 @@
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.netbeans.modules.java.hints.suggestions;
+
+import com.sun.source.tree.BlockTree;
+import com.sun.source.tree.ExpressionTree;
+import com.sun.source.tree.NewClassTree;
+import com.sun.source.tree.ParameterizedTypeTree;
+import com.sun.source.tree.Tree;
+import com.sun.source.tree.Tree.Kind;
+import com.sun.source.tree.VariableTree;
+import com.sun.source.util.TreePath;
+import java.util.List;
+import javax.lang.model.type.TypeMirror;
+import org.netbeans.api.java.queries.SourceLevelQuery;
+import org.netbeans.api.java.source.CompilationInfo;
+import org.netbeans.api.java.source.TreeMaker;
+import org.netbeans.api.java.source.WorkingCopy;
+import org.netbeans.spi.editor.hints.ErrorDescription;
+import org.netbeans.spi.java.hints.ErrorDescriptionFactory;
+import org.netbeans.spi.java.hints.Hint;
+import org.netbeans.spi.java.hints.HintContext;
+import org.netbeans.spi.java.hints.JavaFix;
+import org.netbeans.spi.java.hints.JavaFix.TransformationContext;
+import org.netbeans.spi.java.hints.TriggerPattern;
+import org.openide.util.NbBundle.Messages;
+import org.netbeans.spi.editor.hints.Fix;
+import org.openide.filesystems.FileObject;
+import org.openide.modules.SpecificationVersion;
+import org.netbeans.spi.java.hints.TriggerTreeKind;
+import com.sun.tools.javac.tree.JCTree;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * Hint will convert explicit type to 'var'. Supported: JDK 10 or above
+ *
+ * @author arusinha
+ */
+public class ConvertToVarHint {
+
+    private static final SpecificationVersion JDK_10 = new SpecificationVersion("10"); //NOI18N
+
+    private final static Set<Tree.Kind> LITERALS_TYPE_SET = Collections.unmodifiableSet(
+            new HashSet<>(Arrays.asList(
+                    Tree.Kind.INT_LITERAL,
+                    Tree.Kind.LONG_LITERAL,
+                    Tree.Kind.FLOAT_LITERAL,
+                    Tree.Kind.DOUBLE_LITERAL,
+                    Tree.Kind.BOOLEAN_LITERAL,
+                    Tree.Kind.CHAR_LITERAL,
+                    Tree.Kind.STRING_LITERAL
+            )));
+
+    /**
+     *
+     * @param ctx : HintContext
+     * @return ErrorDescription in case Object reference type and Object
+     * instance type are same.
+     */
+    @Hint(displayName = "#DN_CanUseVarForObjectRef", description = "#DESC_CanUseVarForObjectRef", category = "suggestions") //NOI18N
+
+    @TriggerPattern("$mods$ $type $var =new $type($params$);") //NOI18N
+    @Messages("MSG_ConvertibleToVarType=Explict type can be replaced with 'var'")  //NOI18N
+    public static ErrorDescription checkNewObjInit(HintContext ctx) {
+
+        TreePath treePath = ctx.getVariables().get("$var");
+
+        if (treePath == null) {
+            return null;
+        }
+
+        if (!isHintEnabled(ctx)) {
```

Comment:
Shouldn't isHintEnabled be the first check?

Reply:
this is addressed in commit : https://github.com/apache/incubator-netbeans/pull/463/commits/b733846a4ab3081e2b4d56531ab3c3d041fdd93e